### PR TITLE
Refactor/2318/migrate dialog service

### PR DESCRIPTION
**Pull request by shaman-apprentice**
This PR is build upon https://github.com/MaibornWolff/codecharta/pull/3068 and is ready for review besides changelog

TODO
- [x] merge #3068 first and rebase
- [x] add changelog

### CODE REVIEW

**Review by Hall-Ma — APPROVED** [review]
LGTM!

2 Questions: 
Is it intentional that you no longer have a loading icon in the background when the error dialogue is displayed?

I really like that the 'OK' button has a on hovering shadow in the error dialog but could we change the on hovering color to a darker grey? That would fit better to our ui.

**Comment by shaman-apprentice** [comment]
>  Is it intentional that you no longer have a loading icon in the background when the error dialogue is displayed?

Good catch, not really, but I like it :) The "uploading" has finished as there was an error. Therefore we don't need to wait for the closing of the modal before hiding the loading screen, do we?
 
> I really like that the 'OK' button has a on hovering shadow in the error dialog but could we change the on hovering color to a darker grey? That would fit better to our ui.

The ErrorDialog looks now everywhere the same. Feel free to create a FR for it, I have no opinion there ;)

File: visualization/app/codeCharta/util/loadFilesValidationToErrorDialog.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	const filesWithWarnings = fileValidationResults.filter(validationResult => validationResult.warnings.length > 0)
+	if (filesWithWarnings.length > 0) {
+		htmlMessages.push("<h2>Warnings</h2>")
+		for (const fileWithWarnings of filesWithWarnings) {
+			const fileWarningMessage = buildFileWarningMessage(fileWithWarnings)
+			htmlMessages.push(fileWarningMessage)
+		}
+	}
```
Comment (ce-bo) [inline visualization/app/codeCharta/util/loadFilesValidationToErrorDialog.ts:22]: NIT: It seems that we could extract the two blocks to separate methods as a short refactoring.

**Review by ce-bo — APPROVED** [review]
LGTM!

File: visualization/app/codeCharta/util/loadFilesValidationToErrorDialog.ts (hunk 1 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/util/loadFilesValidationToErrorDialog.ts]: done

**Pull request merged**